compiler/rustc_session/src/config/cfg.rs

```diff
@@ -115,6 +115,18 @@ pub(crate) fn default_configuration(sess: &Session) -> Cfg {
         ins_none!(sym::debug_assertions);
     }
 
+    match sess.opts.unstable_opts.debug_fmt_detail {
+        DebugFmtDetail::Full => {
+            ins_str!(sym::debug_fmt_detail, "full");
```

Comment:
Those cfgs also need to be in the `CheckCfg::fill_well_known` list below (and related files).

Probably like this:
```rust
ins!(sym::debug_fmt_detail, empty_values).extend(&[sym::full, sym::shallow, sym::none]);
```

For more details instructions look at the instructions [at the top of this file](https://github.com/rust-lang/rust/blob/7437d70dd7047c1380fd74ed32416a7c1c690bbe/compiler/rustc_session/src/config/cfg.rs#L10-L20).